src/Parser/Shaman/Elemental/Modules/Items/SmolderingHeart.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import ITEMS from 'common/ITEMS';
+import SPELLS from 'common/SPELLS';
+import ItemIcon from 'common/ItemIcon';
+import ItemLink from 'common/ItemLink';
+import { formatPercentage } from 'common/format';
+
+class SmolderingHeart extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  SmolderingProcs = 0;
+  SmolderingDuration = 0;
+  MaelstromSpent = 0;
+  AscendanceCastEventIds = new Set();
+
+  smoldering_uptime() {
```

Comment:
Both uptime functions can be made into getters
```jsx
get smolderingUptime() {
   return this.smolderingDuration / this.owner.fightDuration;
}
```
and used as `this.smolderingUptime`.